kanidm_client/tests/default_entries.rs

```diff
@@ -0,0 +1,622 @@
+    rsclient.idm_account_create(id, "Deeeeemo").unwrap();
+
+    // Create group and add to user to test read attr: member_of
+    match rsclient.idm_group_get(&group_name).unwrap() {
+        Some(_) => (),
+        None => rsclient.idm_group_create(&group_name).unwrap(),
+    }
+
+    rsclient
+        .idm_group_add_members(&group_name, vec![id])
+        .unwrap();
+}
+
+fn is_attr_writable(rsclient: &KanidmClient, id: &str, attr: &str) -> Option<bool> {
+    println!("writing to attribute: {}", attr);
+    match attr {
+        "radius_secret" => Some(
+            rsclient
+                .idm_account_radius_credential_regenerate(id)
+                .is_ok(),
+        ),
+        "primary_credential" => Some(
+            rsclient
+                .idm_account_primary_credential_set_password(id, "dsadjasiodqwjk12asdl")
+                .is_ok(),
+        ),
+        "ssh_publickey" => Some(
+            rsclient
+                .idm_account_post_ssh_pubkey(
+                    id,
+                    "k1",
+                    "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIAeGW1P6Pc2rPq0XqbRaDKBcXZUPRklo0\
+                L1EyR30CwoP william@amethyst",
+                )
+                .is_ok(),
+        ),
+        "unix_password" => Some(
+            rsclient
+                .idm_account_unix_cred_put(id, "dsadjasiodqwjk12asdl")
+                .is_ok(),
+        ),
+        entry => {
+            let new_value = match entry {
+                "acp_receiver" => "{\"Eq\":[\"memberof\",\"00000000-0000-0000-0000-000000000011\"]}".to_string(),
+                "acp_targetscope" => "{\"And\": [{\"Eq\": [\"class\",\"access_control_profile\"]}, {\"AndNot\": {\"Or\": [{\"Eq\": [\"class\", \"tombstone\"]}, {\"Eq\": [\"class\", \"recycled\"]}]}}]}".to_string(),
+                 _ => id.to_string(),
+            };
+            let m = ModifyList::new_list(vec![
+                Modify::Purged(attr.to_string()),
+                Modify::Present(attr.to_string(), new_value),
+            ]);
+            let f = Filter::Eq("name".to_string(), id.to_string());
+            Some(rsclient.modify(f.clone(), m.clone()).is_ok())
+        }
+    }
+}
+
+fn add_all_attrs(mut rsclient: &mut KanidmClient, id: &str, group_name: &str) {
+    // Extend with posix attrs to test read attr: gidnumber and loginshell
+    rsclient
+        .idm_group_add_members("idm_admins", vec!["admin"])
+        .unwrap();
+    rsclient
+        .idm_account_unix_extend(id, None, Some(&"/bin/bash"))
+        .unwrap();
+    rsclient.idm_group_unix_extend(&group_name, None).unwrap();
+
+    // Extend with person to allow legalname
+    rsclient.idm_account_person_extend(id).unwrap();
+
+    ["ssh_publickey", "legalname", "mail"]
+        .iter()
+        .for_each(|attr| {
+            assert!(is_attr_writable(&rsclient, id, attr).unwrap());
+        });
+
+    // Write radius credentials
+    if id != "anonymous" {
```

Comment:
Could it be worth testing that anonymous gets an error in this path since anon should never have a radius cred?

Reply:
This function is used to add all possible attributes to an account, if that account is anonymous we cannot add radius credentials but who it is a helper function to populate accounts with attributes we cannot fail.

Reply:
My point is more that if the account is anonymous we *expect* to see the failure, so we have the function return a result or similar if we expect the addition to succeed OR fail. This way we are both testing that attributes can be added by who-we-expect but also that they can't be added by accounts we don't expect.

Reply:
Ok, but I how could I populate all attributes to `anonymous` without using that function? I will need to refactor in a way that I can't see. Could you make an example please?

Reply:
Well, rsclient has the associated auth session. So you could have  rsclient do an auth_anonymous, then you have your test just call create_user_with_all_attrs, and you expect a failure instead, so create_user could return a result, and then your test expects on that instead? 

I'm not saying you populate attributes to anonymous, I'm saying you authenticate AS anonymous and attempt to populat the attributes on an account.

Reply:
Line 591. Anonymous tests are at the end.

Reply:
ARRGHH how did I miss this! Okay ignore me :)